src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/list.tsx

```diff
@@ -1,57 +1,145 @@
 import React from 'react';
 import styled from '@emotion/styled';
+import {
+  List,
+  ListRowProps,
+  CellMeasurer,
+  CellMeasurerCache,
+  AutoSizer,
+} from 'react-virtualized';
 
-import space from 'app/styles/space';
-
+import {aroundContentStyle} from './styles';
 import ListHeader from './listHeader';
 import ListBody from './listBody';
-import {aroundContentStyle} from './styles';
+import {BreadcrumbsWithDetails} from './types';
+
+const LIST_MAX_HEIGHT = 400;
 
 type Props = {
   onSwitchTimeFormat: () => void;
-} & Omit<React.ComponentProps<typeof ListBody>, 'relativeTime'>;
+  breadcrumbs: BreadcrumbsWithDetails;
+  relativeTime: string;
+} & Omit<React.ComponentProps<typeof ListBody>, 'breadcrumb' | 'isLastItem' | 'column'>;
 
-const List = React.forwardRef(
-  (
-    {
-      displayRelativeTime,
-      onSwitchTimeFormat,
-      orgId,
+const cache = new CellMeasurerCache({
+  fixedWidth: true,
+  minHeight: 42,
+});
+
+class ListContainer extends React.Component<Props> {
+  componentWillReceiveProps() {
```

Comment:
This hook is deprecated would we be able to use the constructor and componentDidUpdate?

Reply:
DidMount might be better than constructor

Reply:
yeah...I completely forgot that the `componentWillReceiveProps` is now `getDerivedStateFromProps`